### DISCUSSION TITLE
get "volume name" and "Volume Identifier"

### DISCUSSION THREAD

**Discussion by johndanielk** [dell/PyU4V#198] [Show and tell]
HI 
I am working with the application RESTclient10
https://rawstorage.wordpress.com/2024/08/30/undocumented-feature-of-the-unisphere-for-powermax-rest-client-when-working-with-enhanced-api-calls-and-how-to-work-around/

and so far so good - thank you! 
my problem 
I try to find the correct value  "Volume Name" under  "Storage Groups"  > Volumes > Spasific Volume 
and I need to find the "Volume Name" and "Volume Identifier"
but I cannot found on the application RESTclient10 

[![get volume](https://github.com/user-attachments/assets/eff9e761-bccd-4807-ab4b-fc404d368c34)
](url)

**Comment by rawstorage** [comment]
@johndanielk the Volume Name and Identifier are the same thing, the volume.id field is the Hexidecimal system volume number, but yu can select and filter the additional attributes using the API call as shown below,  Note I have 2 volumes in my storage group but I hva only set the identifier on one.

https://ipaddress:8443/univmax/rest/v1/systems/000220200226/volumes?select=id,identifier&filter=storage_groups.id eq audit_test

{
  "volumes": [
    {
      "id": "002C5",
      "identifier": ""
    },
    {
      "id": "00270",
      "identifier": "Test_00270"
    }
  ]
}
Below are couple of screenshots showing select and filter options.
![image](https://github.com/user-attachments/assets/8e576152-f0a1-4435-a499-6f1c1368a7cd)
![image](https://github.com/user-attachments/assets/2541065d-6fe7-4663-baf0-b37d3f9fd597)

**Comment by johndanielk** [comment]
WOW - its works  
Thank you 
base on this URL 
// https://1.1.1.1:8443/univmax/rest/v1/systems/11111111111/volumes?select=id,identifier,wwn,effective_wwn&filter=storage_groups.id eq myStorageGroup

this is my script with Javascript if some one needs


```
function getVolumesByStorageGroup(restHost, symmetrixId, storageGroupId) {
    var filter = "storage_groups.id eq " + storageGroupId;
    var url = "/univmax/rest/v1/systems/" + symmetrixId + "/volumes" +
              "?select=" + encodeURIComponent("id,identifier,effective_wwn") +
              "&filter=" + encodeURIComponent(filter);


    var req = restHost.createRequest("GET", url, null);

    var authHeader = "Basic " + basicAuto; // Add your basic auth token
    req.setHeader("Authorization", authHeader);
    req.setHeader("Accept", "application/json");
    req.setHeader("Content-Type", "application/json");

    var res = req.execute();


    if (res.statusCode != 200) {
        System.log("~~~~ Error: Couldn't retrieve volume details ~~~");
        throw("Couldn't retrieve volume details: " + res.contentAsString);
    }

    var responseJson = JSON.parse(res.contentAsString);
    return responseJson;
}

var symmetrixId = symetrics;  // Replace with your Symmetrix ID
var storageGroupId = "myStorageGroup";  // Replace with your storage group ID

var volumeDetails = getVolumesByStorageGroup(restHost, symmetrixId, storageGroupId);
System.log("Volume Details: " + JSON.stringify(volumeDetails));

```

**Reply by rawstorage** [reply]
No problem, it's an enterprise product after all  :)

If you get in the habit of using the new APIs for these types of lookups as it's much simpler than how we used to do things once you get into the swing of it.
